clients/src/main/java/org/apache/kafka/clients/producer/internals/SenderMetricsRegistry.java

```diff
@@ -16,134 +16,170 @@
+        this.requestLatencyAvg = createMetricName("request-latency-avg", METRIC_GROUP_NAME, "The average request latency in ms", tags);
+        this.requestLatencyMax = createMetricName("request-latency-max", METRIC_GROUP_NAME, "The maximum request latency in ms", tags);
+        this.recordSendRate = createMetricName("record-send-rate", METRIC_GROUP_NAME, "The average number of records sent per second.", tags);
+        this.recordSendTotal = createMetricName("record-send-total", METRIC_GROUP_NAME, "The total number of records sent.", tags);
+        this.recordsPerRequestAvg = createMetricName("records-per-request-avg", METRIC_GROUP_NAME, "The average number of records per request.", tags);
+        this.recordRetryRate = createMetricName("record-retry-rate", METRIC_GROUP_NAME, "The average per-second number of retried record sends", tags);
+        this.recordRetryTotal = createMetricName("record-retry-total", METRIC_GROUP_NAME, "The total number of retried record sends", tags);
+        this.recordErrorRate = createMetricName("record-error-rate", METRIC_GROUP_NAME, "The average per-second number of record sends that resulted in errors", tags);
+        this.recordErrorTotal = createMetricName("record-error-total", METRIC_GROUP_NAME, "The total number of record sends that resulted in errors", tags);
+        this.recordSizeMax = createMetricName("record-size-max", METRIC_GROUP_NAME, "The maximum record size", tags);
+        this.recordSizeAvg = createMetricName("record-size-avg", METRIC_GROUP_NAME, "The average record size", tags);
+        this.requestsInFlight = createMetricName("requests-in-flight", METRIC_GROUP_NAME, "The current number of in-flight requests awaiting a response.", tags);
+        this.metadataAge = createMetricName("metadata-age", METRIC_GROUP_NAME, "The age in seconds of the current producer metadata being used.", tags);
+        this.batchSplitRate = createMetricName("batch-split-rate", METRIC_GROUP_NAME, "The average number of batch splits per second", tags);
+        this.batchSplitTotal = createMetricName("batch-split-total", METRIC_GROUP_NAME, "The total number of batch splits", tags);
+
+        this.produceThrottleTimeAvg = createMetricName("produce-throttle-time-avg", METRIC_GROUP_NAME, "The average time in ms a request was throttled by a broker", tags);
+        this.produceThrottleTimeMax = createMetricName("produce-throttle-time-max", METRIC_GROUP_NAME, "The maximum time in ms a request was throttled by a broker", tags);
 
         /* ***** Topic level *****/
-        Set<String> topicTags = new HashSet<String>(tags);
-        topicTags.add("topic");
+        this.topicTags = new HashSet<String>(tags);
+        this.topicTags.add("topic");
+
+        this.topicRecordSendRate = createTemplate("record-send-rate", TOPIC_METRIC_GROUP_NAME, "The average number of records sent per second for a topic.", topicTags);
```

Comment:
Worth adding a comment explaining why we can't create the `metricName` for these (i.e. we don't know the topic name).